src/archive/rar.rs

```diff
@@ -35,17 +47,22 @@ pub fn unpack_archive(archive_path: &Path, output_folder: &Path, quiet: bool) ->
 }
 
 /// List contents of `archive_path`, returning a vector of archive entries
-pub fn list_archive(archive_path: &Path) -> impl Iterator<Item = crate::Result<FileInArchive>> {
-    Archive::new(archive_path)
-        .open_for_listing()
-        .expect("cannot open archive")
-        .map(|item| {
-            let item = item?;
-            let is_dir = item.is_directory();
-            let path = item.filename;
-
-            Ok(FileInArchive { path, is_dir })
-        })
+pub fn list_archive(
+    archive_path: &Path,
+    password: Option<impl AsRef<[u8]>>,
+) -> impl Iterator<Item = crate::Result<FileInArchive>> {
+    let password = password.as_ref().map(|p| p.as_ref());
+    let archive = match password {
+        Some(password) => Archive::with_password(archive_path, password),
+        None => Archive::new(archive_path),
+    };
+    archive.open_for_listing().expect("cannot open archive").map(|item| {
```

Comment:
This was already present, but please replace this `expect` by a `?`

Reply:
done